File: ezSCORES/ezSCORES.Model/Rewards.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ezSCORES.Model
{
	public class Rewards
	{
		public int Id { get; set; }

		public int CompetitionId { get; set; }

		public string Name { get; set; } = null!;

		public int? RankingPosition { get; set; }

		public int Amount { get; set; }

		public string? Description { get; set; }
	}
}

File: ezSCORES/UI/ezscores_desktop/lib/models/rewards.dart
import 'package:json_annotation/json_annotation.dart';

part 'rewards.g.dart';

@JsonSerializable()
class Rewards {
	int? id;
	int? competitionId;
	String? name;
	int? rankingPosition;
	int? amount;
	String? description;

	Rewards();

	factory Rewards.fromJson(Map<String, dynamic> json) => _$RewardsFromJson(json);

	Map<String, dynamic> toJson() => _$RewardsToJson(this);
}
